File: PortalOS/PortalOS.Domain/Services/ClienteService.cs
using dNET.Domain.Repositories;
using dNET.Domain.Services;
using PortalOS.Domain.Entities;
using PortalOS.Domain.ViewModels;

namespace PortalOS.Domain.Services
{
    public class ClienteService : ServiceCrud<Cliente>
    {
        public ClienteService(IUnitOfWork unitOfWork) : base(unitOfWork)
        {
        }

        public override bool CustomValidate(Cliente entity)
        {
            if (!string.IsNullOrEmpty(entity.Cnpj) && Exists(c => c.Cnpj == entity.Cnpj && c.Id != entity.Id))
            {
                return MessageError("Cliente com este CNPJ ja existe");
            }
            return base.CustomValidate(entity);
        }

        public Cliente Create(CreateClienteRequest request)
        {
            var cliente = new Cliente
            {
                RazaoSocial = request.RazaoSocial,
                Cnpj = request.Cnpj,
                Responsavel = request.Responsavel,
                Telefone = request.Telefone,
                EmailResponsavel = request.EmailResponsavel,
                EmailFinanceiro = request.EmailFinanceiro,
                Endereco = request.Endereco,
                StatusCliente = request.StatusCliente,
                ClienteTotvs = request.ClienteTotvs
            };

            Insert(cliente);
            return cliente;
        }

        public Cliente Update(long id, UpdateClienteRequest request)
        {
            if (id != request.Id)
            {
                ThrowError("ID da URL nao confere com ID do body");
            }

            var cliente = GetById(id);
            if (cliente == null)
            {
                ThrowError("Cliente nao encontrado");
            }

            cliente.RazaoSocial = request.RazaoSocial;
            cliente.Cnpj = request.Cnpj;
            cliente.Responsavel = request.Responsavel;
            cliente.Telefone = request.Telefone;
            cliente.EmailResponsavel = request.EmailResponsavel;
            cliente.EmailFinanceiro = request.EmailFinanceiro;
            cliente.Endereco = request.Endereco;
            cliente.StatusCliente = request.StatusCliente;
            cliente.ClienteTotvs = request.ClienteTotvs;

            Merge(cliente);
            return cliente;
        }

        public IEnumerable<Cliente> GetAtivos()
        {
            return Query(c => c.StatusCliente == ValueObjects.StatusCliente.Ativo);
        }
    }
}

File: PortalOS/PortalOS.Web/src/services/clienteService.ts
import { httpClient, ODataHelper } from 'd-rts';
import type { Cliente, CreateClienteRequest, UpdateClienteRequest } from '../types/cliente';
import type { PaginatedResult, PaginationParams } from 'd-rts';

const BASE_URL = '/cliente';

export const clienteService = {
  getAll: async (params?: PaginationParams): Promise<PaginatedResult<Cliente>> => {
    const oDataParams = params ? ODataHelper.fromPaginationParams(params) : { $count: true };

    if (params?.search) {
      oDataParams.$filter = ODataHelper.createSearchFilter(params.search, ['razaoSocial', 'cnpj', 'responsavel']);
    }

    oDataParams.$select = 'Id,RazaoSocial,Cnpj,Responsavel,Telefone,EmailResponsavel,EmailFinanceiro,Endereco,StatusCliente,ClienteTotvs';

    const query = ODataHelper.buildQuery(oDataParams);
    const response = await httpClient.get<Cliente[]>(`${BASE_URL}/GetAll${query}`);

    return ODataHelper.processResponse<Cliente>(response.data, params);
  },

  getById: (id: number) => httpClient.get<Cliente>(`${BASE_URL}/${id}`),

  getAtivos: () => httpClient.get<Cliente[]>(`${BASE_URL}/ativos`),

  create: (data: CreateClienteRequest) => httpClient.post<Cliente>(`${BASE_URL}/Create`, data),

  update: (id: number, data: UpdateClienteRequest) => httpClient.put<Cliente>(`${BASE_URL}/${id}`, data),

  delete: (id: number) => httpClient.delete(`${BASE_URL}/${id}`),
};
